examples/jsm/renderers/webgl/WebGLBackend.js

```diff
@@ -350,55 +354,111 @@ class WebGLBackend extends Backend {
 
 	beginCompute( /*computeGroup*/ ) {
 
-		console.warn( 'Abstract class.' );
+		const gl = this.gl;
+
+		gl.bindFramebuffer( gl.FRAMEBUFFER, null );
 
 	}
 
-	compute( /*computeGroup, computeNode, bindings, pipeline*/ ) {
+	compute( computeGroup, computeNode, bindings, pipeline ) {
 
-		console.warn( 'Abstract class.' );
+		const gl = this.gl;
+
+		if ( ! this.discard )  {
+
+			// required here to handle async behaviour of render.compute()
+			gl.enable( gl.RASTERIZER_DISCARD );
+			this.discard = true;
+		}
+
+		const { programGPU, transformBuffers, attributes } = this.get( pipeline );
+
+		const vaoKey = this._getVaoKey( null, attributes );
+
+		let vaoGPU = this.vaoCache[ vaoKey ];
+
+		if ( vaoGPU === undefined ) {
+
+			this._createVao( null, attributes );
+
+		} else {
+
+			gl.bindVertexArray( vaoGPU );
+
+		}
+
+		gl.useProgram( programGPU );
+
+		this._bindUniforms( bindings );
+
+		const transformFeedbackGPU = this._getTransformFeedback( transformBuffers );
+
+		gl.bindTransformFeedback( gl.TRANSFORM_FEEDBACK, transformFeedbackGPU );
+		gl.beginTransformFeedback( gl.POINTS );
+
+		gl.drawArraysInstanced( gl.POINTS, 0, 1, computeNode.count );
+
+		gl.endTransformFeedback();
+		gl.bindTransformFeedback( gl.TRANSFORM_FEEDBACK, null );
+
+		// switch active buffers
+		for ( let i = 0; i < transformBuffers.length; i ++ ) {
+
+			transformBuffers[ i ].switchBuffers();
+
+		}
 
 	}
 
 	finishCompute( /*computeGroup*/ ) {
 
-		console.warn( 'Abstract class.' );
+		const gl = this.gl;
+
+		this.discard = false;
```

Comment:
Do we really need this flag? Maybe RASTERIZER_DISCARD can be just always enabled in beginCompute and disabled in finishCompute? /ping @aardgoose @sunag

Reply:
I did have it that way at first, but for some reason which I can't remember that had issues. If I get a chance I will try and reproduce the problem.

Reply:
Checked again, the check deals with nested compute shaders.

Reply:
Hm... Feels a bit strange for me. So you mean one shader starts executing before another is finished? Is this intended?

Reply:
The comment needs fixing. Actually the problem is that compute passes can be nested and indeed are in the compute_points example, which is where I hit this issue:

https://github.com/mrdoob/three.js/blob/bd33e679a5b634842bb39a377b9a37eb68bc0e2a/examples/webgpu_compute_points.html#L121

Reply:
I just thought about this again and I think the current approach is completely equivalent to always calling `enable` in `compute` (the only possibility is that it could be called multiple times -- but I guess it wouldn't break?) and `disable` in `finishCompute` (it already does this). Can you please check this?

Reply:
WIth nested calls you get the sequence, seen in practice in compute_points: I originally had enable in discard in beginCompute().

```
beginCompute()
   beginCompute() <- called from onBeforeRender()
      compute()
   endCompute()
   compute() <- discard disabled by previous endCompute()
endCompute()
```

Reply:
> compute() <- discard disabled by previous endCompute()

No -- I'm suggesting that it be enabled in compute() instead of beginCompute() (currently it is the same).